src/pages/tasks/NewTaskPage.tsx

```diff
@@ -1,141 +1,123 @@
         // this allows us to go ahead and set that report as the share destination
         // and disable the share destination selector
-        if (props.task.parentReportID) {
-            Task.setShareDestinationValue(props.task.parentReportID);
+        if (task?.parentReportID) {
+            TaskActions.setShareDestinationValue(task.parentReportID);
         }
 
         // If we have a share destination, we want to set the parent report and
         // the share destination data
-        if (props.task.shareDestination) {
-            setParentReport(lodashGet(props.reports, `report_${props.task.shareDestination}`, {}));
-            const displayDetails = Task.getShareDestination(props.task.shareDestination, props.reports, props.personalDetails);
+        if (task?.shareDestination) {
+            setParentReport(reports?.[`report_${task.shareDestination}`] ?? null);
+            const displayDetails = TaskActions.getShareDestination(task.shareDestination, reports, personalDetails);
             setShareDestination(displayDetails);
         }
 
         // If we have a title, we want to set the title
-        if (!_.isUndefined(props.task.title)) {
-            setTitle(props.task.title);
+        if (task?.title !== undefined) {
+            setTitle(task.title);
         }
 
         // If we have a description, we want to set the description
-        if (!_.isUndefined(props.task.description)) {
-            setDescription(props.task.description);
+        if (task?.description !== undefined) {
+            setDescription(task.description);
         }
-    }, [props]);
+    }, [personalDetails, reports, task?.assignee, task?.assigneeAccountID, task?.description, task?.parentReportID, task?.shareDestination, task?.title]);
 
     // On submit, we want to call the createTask function and wait to validate
     // the response
-    function onSubmit() {
-        if (!props.task.title && !props.task.shareDestination) {
+    const onSubmit = () => {
+        if (!task?.title && !task?.shareDestination) {
             setErrorMessage('newTaskPage.confirmError');
             return;
         }
 
-        if (!props.task.title) {
+        if (!task?.title) {
             setErrorMessage('newTaskPage.pleaseEnterTaskName');
             return;
         }
 
-        if (!props.task.shareDestination) {
+        if (!task?.shareDestination) {
             setErrorMessage('newTaskPage.pleaseEnterTaskDestination');
             return;
         }
 
         playSound(SOUNDS.DONE);
-        Task.createTaskAndNavigate(
-            parentReport.reportID,
-            props.task.title,
-            props.task.description,
-            props.task.assignee,
-            props.task.assigneeAccountID,
-            props.task.assigneeChatReport,
-            parentReport.policyID,
+        TaskActions.createTaskAndNavigate(
+            parentReport?.reportID ?? '',
+            task.title,
```

Comment:
NAB: It seems weird to not have a default for the title, given that we use optional chaining on the task, with defaults, for other params.

Reply:
I think it is because others except task title are optional here so there is always title field in task and does not need a default value.

Reply:
Yes, but why is the optional chaining operator after the task directly? The task can be null right, and that's why it's there?

I think it's probably fine, but still looking for more of an explanation.

Reply:
Oh, I think we need not optional chain for `description` and `assignee` here. If `task` is `null` or `undefined` this check
```js
if (!task?.title && !task?.shareDestination) {
            setErrorMessage('newTaskPage.confirmError');
            return;
        }
```
returns early anyway.

Reply:
@c3024 removed